### PR TITLE
Specify which params get passed to jax.jacfwd

### PR DESCRIPTION
**Pull request by skhrg**
The time of `jit_conv_int_gnfw` is dominated by the differentiation, time scales roughly linearly with the number of arguments you diff over. 

Note the function now outputs `grad` with the axis in a different order, removing the need for `derivs = jnp.moveaxis(derivs, 2, 0)` in a helper function.

To specify which arguments to use pass a tuple with the argument numbers as `argnums`. This can be done in a helper function like this:

```
def helper(params, tod):
      x = tod.info['dx']
      y = tod.info['dy']

      xy = [x, y]
      xy = jnp.asarray(xy)
       
     argnums = [i for i, p in enumerate(to_fit[:len(params)]) if p]
     pred, derivs = jit_conv_int_gnfw_elliptical(params, xy, z, argnums=tuple(argnums))

    return derivs, pred
```

This function assumes that `to_fit` is declared before the function is called (not currently the case in `ms0735.py` but fixing this just requires moving its declaration up)

### CODE REVIEW

**Comment by skhrg** [comment]
On second thought a better way of doing the `helper` function would be something like 

```
def helper(params, tod, z, to_fit):
      x = tod.info['dx']
      y = tod.info['dy']

      xy = [x, y]
      xy = jnp.asarray(xy)
       
     argnums = [i for i, p in enumerate(to_fit[:len(params)]) if p]
     pred, derivs = jit_conv_int_gnfw_elliptical(params, xy, z, argnums=tuple(argnums))

    return derivs, pred
```

And then passing `partial(helper, z=z, to_fit=to_fit)` to `minkasi` as needed. This requires `from functools import partial`
